app/src/main/java/org/mozilla/focus/autocomplete/AutocompleteRemoveFragment.kt

```diff
@@ -28,14 +29,16 @@ class AutocompleteRemoveFragment : AutocompleteListFragment() {
     }
 
     private fun removeSelectedDomains(context: Context) {
-        launch(CommonPool) {
-            val domains = (domainList.adapter as DomainListAdapter).selection()
-            if (!domains.isEmpty()) {
-                CustomAutocomplete.removeDomains(context, domains)
+        val domains = (domainList.adapter as DomainListAdapter).selection()
+        if (domains.isNotEmpty()) {
+            launch(UI) {
+                async {
+                    CustomAutocomplete.removeDomains(context, domains)
```

Comment:
I see, this async/await is capturing the context before popping the backstack, and then finishing it on a background thread.